source/tests/pt/test_weighted_avg.py

```diff
@@ -0,0 +1,451 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+import unittest
+
+import numpy as np
+
+from deepmd.entrypoints.test import (
+    ener_err,
+    ener_err_ops,
+)
+
```

Comment:
Note that you should test `test_ener` instead of only testing `ener_err` and  `ener_err_ops`!